test/token_holder/submit_authorize_session.js

```diff
@@ -0,0 +1,556 @@
+                },
+            );
+
+            const keyData = await tokenHolder.ephemeralKeys.call(ephemeralKey);
+
+            assert.isOk(
+                keyData.spendingLimit.eqn(spendingLimit),
+            );
+
+            assert.isOk(
+                keyData.nonce.eqn(0),
+            );
+
+            assert.isOk(
+                keyData.expirationHeight.eqn(expirationHeight),
+            );
+
+            assert.isOk(
+                // TokenHolder.AuthorizationStatus.AUTHORIZED == 1
+                keyData.status.eqn(1),
+            );
+
+            const transaction = await tokenHolder.transactions.call(
+                transactionID,
+            );
+
+            assert.strictEqual(
+                transaction.destination,
+                tokenHolder.address,
+            );
+
+            assert.strictEqual(
+                transaction.data,
+                generateSubmitAuthorizeSessionData(
+                    ephemeralKey, spendingLimit, expirationHeight,
+                ),
+            );
+
+            assert.isOk(
+                transaction.executed,
+            );
+        });
+
+        it('Submit => Confirm', async () => {
+            const required = 2;
+
+            const registeredWallet0 = accounts[0];
+            const registeredWallet1 = accounts[1];
+
+            const wallets = [registeredWallet0, registeredWallet1];
+
+            const brandedToken = accounts[2];
+            const tokenRules = accounts[3];
+
+            const tokenHolder = await TokenHolder.new(
+                brandedToken, tokenRules, wallets, required,
+            );
+
+            const ephemeralKey = '0x62502C4DF73935D0D10054b0Fb8cC036534C6fb0';
+            const spendingLimit = 1;
+            const expirationHeight = await web3.eth.getBlockNumber() + 50;
+
+
+            const transactionID = await TokenHolderHelper.submitAuthorizeSession(
+                tokenHolder,
+                ephemeralKey,
+                spendingLimit,
+                expirationHeight,
+                {
+                    from: registeredWallet0,
+                },
+            );
+
+            const keyData = await tokenHolder.ephemeralKeys.call(ephemeralKey);
+
+            assert.isOk(
+                // TokenHolder.AuthorizationStatus.NOT_AUTHORIZED == 0
+                keyData.status.eqn(0),
+            );
+
+            const transaction = await tokenHolder.transactions.call(
+                transactionID,
+            );
+
+            assert.strictEqual(
+                transaction.destination,
+                tokenHolder.address,
+            );
+
+            assert.strictEqual(
+                transaction.data,
+                generateSubmitAuthorizeSessionData(
+                    ephemeralKey, spendingLimit, expirationHeight,
+                ),
+            );
+
+            assert.isNotOk(
+                transaction.executed,
+            );
+        });
+    });
+});
```

Comment:
More Test Case To Add:
- submitAuthorizeSession with spending limit = 0 should be success.

Reply:
We don't need to check for all valid values. Only invalid ones.